htmresearch/algorithms/faulty_temporal_memory.py

```diff
@@ -52,12 +135,26 @@ def __init__(self,
     self.numDead = 0
 
 
+  @staticmethod
+  def connectionsFactory(*args, **kwargs):
+    """
+    Override connectionsFactory() in base class to return a Connections subclass
+    that supports cell destruction.
+
+    See Connections for constructor signature and usage
+
+    @return: FaultyConnections instance
+    """
+    return FaultyConnections(*args, **kwargs)
+
+
   def killCells(self, percent = 0.05):
```

Comment:
Since this destroys segments on the cell, it should also remove any destroyed segments from `tm.activeSegments` and `tm.matchingSegments`.